trace_decoder/src/processed_block_trace.rs

```diff
@@ -4,272 +4,58 @@ use std::iter::once;
-    tries: HashMap<HashedAccountAddr, SeparateTriePreImage>,
-) -> HashMap<HashedAccountAddr, HashedPartialTrie> {
-    tries
-        .into_iter()
-        .map(|(k, v)| match v {
-            SeparateTriePreImage::Uncompressed(_) => todo!(),
-            SeparateTriePreImage::Direct(t) => (k, t.0),
-        })
-        .collect()
-}
-
-fn process_compact_trie(trie: TrieCompact) -> CompactParsingResult<ProcessedBlockTracePreImages> {
-    let out = process_compact_prestate_debug(trie)?;
-
-    if !COMPATIBLE_HEADER_VERSIONS
-        .iter()
-        .any(|&v| out.header.version_is_compatible(v))
-    {
-        return Err(CompactParsingError::IncompatibleVersion(
-            COMPATIBLE_HEADER_VERSIONS.to_vec(),
-            out.header.version,
-        ));
-    }
-
-    Ok(out.into())
+pub(crate) struct ProcessedBlockTrace {
+    pub tries: PartialTriePreImages,
+    pub txn_info: Vec<ProcessedTxnInfo>,
+    pub withdrawals: Vec<(Address, U256)>,
 }
 
-/// Structure storing a function turning a `CodeHash` into bytes.
 #[derive(Debug)]
-pub struct ProcessingMeta<F>
-where
-    F: CodeHashResolveFunc,
-{
-    resolve_code_hash_fn: F,
-}
-
-impl<F> ProcessingMeta<F>
-where
-    F: CodeHashResolveFunc,
-{
-    /// Returns a `ProcessingMeta` given the provided code hash resolving
-    /// function.
-    pub const fn new(resolve_code_hash_fn: F) -> Self {
-        Self {
-            resolve_code_hash_fn,
-        }
-    }
+pub(crate) struct ProcessedBlockTracePreImages {
+    pub tries: PartialTriePreImages,
+    pub extra_code_hash_mappings: Option<HashMap<H256, Vec<u8>>>,
 }
 
 #[derive(Debug)]
 pub(crate) struct ProcessedTxnInfo {
-    pub(crate) nodes_used_by_txn: NodesUsedByTxn,
-    pub(crate) contract_code_accessed: HashMap<CodeHash, Vec<u8>>,
-    pub(crate) meta: TxnMetaState,
+    pub nodes_used_by_txn: NodesUsedByTxn,
+    pub contract_code_accessed: HashMap<H256, Vec<u8>>,
+    pub meta: TxnMetaState,
 }
 
-struct CodeHashResolving<F> {
+pub(crate) struct CodeHashResolving<F> {
     /// If we have not seen this code hash before, use the resolve function that
     /// the client passes down to us. This will likely be an rpc call/cache
     /// check.
-    client_code_hash_resolve_f: F,
+    pub client_code_hash_resolve_f: F,
 
     /// Code hash mappings that we have constructed from parsing the block
     /// trace. If there are any txns that create contracts, then they will also
     /// get added here as we process the deltas.
-    extra_code_hash_mappings: HashMap<CodeHash, Vec<u8>>,
+    pub extra_code_hash_mappings: HashMap<H256, Vec<u8>>,
 }
 
-impl<F: CodeHashResolveFunc> CodeHashResolving<F> {
-    fn resolve(&mut self, c_hash: &CodeHash) -> Vec<u8> {
+impl<F: Fn(&H256) -> Vec<u8>> CodeHashResolving<F> {
+    fn resolve(&mut self, c_hash: &H256) -> Vec<u8> {
         match self.extra_code_hash_mappings.get(c_hash) {
             Some(code) => code.clone(),
             None => (self.client_code_hash_resolve_f)(c_hash),
```

Comment:
Non-blocking: Would we ever really need to dynamically retrieve the hash or we could always get it before the trace decoder starts working? We would probably have to spin up tokio async engine instance and combine multiple resolve calls with this one engine

Reply:
We could, but we would need to do a separate pass over the block trace and all of the compact, which I think will be a bit slower (but it's already pretty fast anyways I guess). Right now, this resolve function is never actually hit. We always have the matching code for all code hashes embedded inside of the compact encoding pre-images (with the exception of when contract creation occurs).

Reply:
As discussed in slack, my short-term goal is to remove this. we can add it back when it's needed.